server/handler_test.go

```diff
@@ -49,6 +54,185 @@ func TestLSPHandler(t *testing.T) {
 	}
 }
 
+func TestConcurrentMethods(t *testing.T) {
+	wg := sync.WaitGroup{}
+	mu := sync.Mutex{}
+	received := []string{}
+	processed := make(chan string, 20)
+
+	// This handler will sleep for different durations based on the method
+	// This allows us to validate that concurrent methods don't block each other
+	mockHandler := &MockHandler{
+		handler: func(ctx context.Context, conn *jsonrpc2.Conn, request *jsonrpc2.Request) {
+			if request.Method == "concurrent-0" {
+				// Make the first concurrent method take longer
```

Comment:
probably ok here, but generally try to avoid using time to order things in tests:
- more likely you'll end up writing race conditions, which means your test will fail when run with `-race` 
- more likely to flake
- takes longer to run

instead, better to use things like wait-groups or synchronisation.

Reply:
A good point, I changed it for the channels approach you mentioned below, not it should be much more stable and readable. Thanks!